core/cloudflow-maven-plugin/src/main/scala/com/lightbend/cloudflow/maven/CloudflowAggregator.scala

```diff
@@ -0,0 +1,171 @@
+package com.lightbend.cloudflow.maven
+
+import cloudflow.blueprint.deployment.CloudflowCRFormat._
+import cloudflow.blueprint.deployment._
+import cloudflow.cr.Generator
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.apache.maven.artifact.Artifact
+import org.apache.maven.plugin.logging.Log
+import org.apache.maven.project.MavenProject
+import spray.json._
+
+import java.io.File
+import java.net.{ URL, URLEncoder }
+import java.nio.file.Paths
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object CloudflowAggregator {
+
+  private def findBlueprint(project: MavenProject) = {
+    val blueprintFile =
+      Paths.get(project.getBasedir.getAbsolutePath, "src", "main", "blueprint", "blueprint.conf").toFile
+    if (blueprintFile.exists()) {
+      Some(blueprintFile)
+    } else {
+      None
+    }
+  }
+
+  def getBlueprint(allProjects: Seq[MavenProject], log: Log): Option[File] = {
+    allProjects.foldLeft(Option.empty[File]) {
+      case (acc, project) =>
+        log.info(s"checking blueprint in $project")
+
+        findBlueprint(project).orElse(acc)
+    }
+  }
+
+  def getStreamlets(allProjects: Seq[MavenProject], log: Log) = {
+    allProjects.foldLeft(Map.empty[String, Map[String, Config]]) {
+      case (acc, project) =>
+        try {
+
+          log.info(s"extracting streamlets $project")
+
+          val rawStreamlets =
+            FileUtil.readLines(new File(project.getBuild.getDirectory, Constants.STREAMLETS_FILE))
+
+          val streamlets: Map[String, Config] = rawStreamlets.map { k =>
+            val file = new File(project.getBuild.getDirectory, URLEncoder.encode(k, "UTF-8"))
+            val config = ConfigFactory
+              .parseFile(file)
+              .resolve()
+            k -> config
+          }.toMap
+
+          (acc + (project.getName -> streamlets))
+        } catch {
+          case ex: Throwable =>
+            log.warn(s"No streamlets found in project $project")
+            (acc)
+        }
+    }
+  }
+
+  def getImages(allProjects: Seq[MavenProject], log: Log) = {
+    allProjects.foldLeft(Map.empty[String, String]) {
+      case (acc, project) =>
+        try {
+
+          log.info(s"extracting image names $project")
+
+          val image =
+            FileUtil.readLines(new File(project.getBuild.getDirectory, Constants.DOCKER_IMAGE_FILE)).mkString.trim
+
+          (acc + (project.getName -> image))
+        } catch {
+          case ex: Throwable =>
+            log.warn(s"No images found in project $project")
+            (acc)
+        }
+    }
+  }
+
+  def getCR(topLevel: MavenProject): Option[CloudflowCR] = {
```

Comment:
nitpick: function return type to be specified

Reply:
ok